Tools/GNUMake/Make.machines

```diff
@@ -72,6 +72,11 @@ ifeq ($(findstring theta, $(host_name)), theta)
   which_computer := theta
 endif
 
+ifeq ($(findstring polaris, $(host_name)), polaris)
```

Comment:
If possible, could we add an extra test of whether this is an ALCF machine similar to `ifdef NERSC_HOST`, `ifdef OLCF_MODULEPATH` etc.?

Reply:
I checked `env` for anything containing `host` `ALCF` `Argonne` as well as looked at the documentation, and I don't see anything likely to check. 

It might be possible to check for Cray variables / modules, but that's not deterministic. I could also check for the presence of various 0-size files and make assumptions based on which lustre / home file systems are available...

Reply:
What does `hostname -f` show?

Reply:
It ends in `polaris.alcf.anl.gov` after some node-name specific type stuff, depending on if it's a login node or a compute node. That's [`host_name := $(shell hostname -f)`
](https://github.com/AMReX-Codes/amrex/blob/development/Tools/GNUMake/Make.machines#L15) in Make.machines, right?

Reply:
Yes.  You can also use `make .... print-host_name` to find out what it is in the make.

Reply:
It looks like I don't have theta access, so I'm unsure about adding a generic ALCF test similar to the NERSC_HOST block since I can't test it.

Reply:
Don't worry about theta.

Reply:
Example ThetaGPU compute node:
```
$ hostname -f
thetagpu01.mcp.alcf.anl.gov
```
Theta login node (shared between CPU and GPU nodes):
```
$ hostname -f
thetalogin4.tmi.alcf.anl.gov
```

Reply:
I've pushed a commit that adds a check for alcf.anl.gov